src/xunit.analyzers/AssertEqualPrecisionShoulBeInRangeFixer.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Xunit.Analyzers
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+    public class AssertEqualPrecisionShoulBeInRangeFixer : CodeFixProvider
+    {
+        private const string title = "Use precision 0";
```

Comment:
I'm not sure if it's a best option we can deliver. At least it's a straightforward one and will fail tricky test cases, so it should force people to think what is the best precision for their cases.